src/org/wso2/tg/jenkins/util/ConfigUtils.groovy

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.tg.jenkins.util
+
+/**
+ * Populates the TestGrid yaml input parameters in infra,deployment,scenario sections
+ * according to the config properties file definitions
+ *
+ * Ex :
+ * common.xxx=yyyy (insert the input parameter to all three sections)
+ * infra.default.xxx=yyy (insert to input parameter section of infra section by default)
+ * infra.xxx=yyy (insert to input parameter section of infra section only if user has specified in the
+ * testgridProvidedParameters section
+ *
+ * @param tgYamlContent parsed testgrid yaml file content
+ * @param commonConfigs parsed common-config  properties file content
+ * @return testgrid yaml file content with input parameters correctly set
+ */
+def addCommonConfigsToTestGridYaml(tgYamlContent, commonConfigs) {
+
+    echo "Populating testgrid yaml file with common configs"
+
+    commonConfigs.each { key, value ->
+        def infraProvisioners = null;
+        def deploymentPatterns = null;
+        def scenarioConfigs = null;
+
+        if (tgYamlContent.infrastructureConfig !=null ){
+            infraProvisioners = tgYamlContent.infrastructureConfig.provisioners
+        }
+        if(tgYamlContent.deploymentConfig!=null){
+            deploymentPatterns = tgYamlContent.deploymentConfig.deploymentPatterns
+        }
+
+        scenarioConfigs = tgYamlContent.scenarioConfigs
+
+        //split each property from . symbol
+        def split = key.split('\\.');
```

Comment:
I think we should split with a limit. We have to assume that the property name itself may contain a dot.